doc/compat/latch.adoc

```diff
@@ -0,0 +1,133 @@
+////
+Copyright 2023 Christian Mazakas
+Distributed under the Boost Software License, Version 1.0.
+https://www.boost.org/LICENSE_1_0.txt
+////
+
+[#latch]
+# Latch
```

Comment:
Section headers should probably be the header names, `<boost/core/latch.hpp>` in this case.